be/src/query/query.service.ts

```diff
@@ -1,24 +1,75 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { Inject, Injectable, NotFoundException } from '@nestjs/common';
 import { QueryDto } from './dto/query.dto';
 import { QUERY_DB_ADAPTER } from '../config/query-database/query-db.moudle';
 import { QueryDBAdapter } from '../config/query-database/query-db.adapter';
-import { ResQueryDto } from './dto/res-query.dto';
+import { QueryType } from '../common/enums/query-type.enum';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Shell } from '../shell/shell.entity';
+import { Repository } from 'typeorm';
 
 @Injectable()
 export class QueryService {
   constructor(
     @Inject(QUERY_DB_ADAPTER) private readonly queryDBAdapter: QueryDBAdapter,
+    @InjectRepository(Shell)
+    private shellRepository: Repository<Shell>,
   ) {}
 
-  async execute(userId: string, queryDto: QueryDto) {
-    const connection = await this.queryDBAdapter.createConnection(userId);
+  async execute(sessionId: string, shellId: number, queryDto: QueryDto) {
+    const shell = await this.shellRepository.findOne({ where: { shellId } });
+    if (!shell) {
+      throw new NotFoundException('존재하지 않는 shellId 입니다.');
+    }
+
+    const connection = await this.queryDBAdapter.createConnection(sessionId);
+    const baseUpdateData = {
+      query: queryDto.query,
+      queryType: this.detectQueryType(queryDto.query),
+    };
+
     try {
       const rows = await this.queryDBAdapter.run(connection, queryDto.query);
-      return ResQueryDto.ok(queryDto.query, rows);
+      const slicedRows = rows.length > 100 ? rows.slice(0, 100) : rows;
+
+      await this.updateShell(shellId, shell, {
+        ...baseUpdateData,
+        affectedRows: rows.length,
+        queryStatus: true,
+        resultTable: slicedRows,
+      });
     } catch (e) {
-      return ResQueryDto.fail(e.sqlMessage);
+      await this.updateShell(shellId, shell, {
+        ...baseUpdateData,
+        queryStatus: false,
+        failMessage: e.sqlMessage,
+      });
     } finally {
       this.queryDBAdapter.closeConnection(connection);
     }
+
+    return shell;
+  }
+
+  private async updateShell(
+    shellId: number,
+    shell: Shell,
+    updateData: Partial<Shell>,
+  ) {
+    await this.shellRepository.update({ shellId }, updateData);
+    Object.assign(shell, updateData);
```

Comment:
이부분 assign 하고 업데이트 하는 부분 순서가 바뀐 것 같아요.
Shell 엔티티도 찾았으니 save 로 해주면 어떨까요? 그러면 저번에 말씀드린 훅도 사용할 수 있을 것 같아요.
```suggestion
    Object.assign(shell, updateData);
    await this.shellRepository.save(shell);

```